src/Remapping/distributed_remapping.jl

```diff
@@ -77,43 +112,77 @@ function target_hcoords_pid_bitmask(target_hcoords, topology, pid)
+
+    comms_ctx::CC
+
+    # Space over which the remapper is defined
+    space::SPACE
 
     # Target points that are on the process where this object is defined
     # local_target_hcoords is stored as a 1D array (we store it as 1D array because in
     # general there is no structure to this object, especially for cubed sphere, which have
     # points spread all over the place)
     local_target_hcoords::T1
 
-    # Target coordinates in the vertical direction. zcoords are the same for all the processes
-    target_zcoords::T2
+    # Target coordinates in the vertical direction. zcoords are the same for all the processes.
+    # target_zcoords are always assumed to be "reference z"s.
+    target_zcoords::TARG_Z
 
     # bitmask that identifies which target points are on process where the object is
     # defined. In general, local_target_points_bitmask is a 2D matrix and it is used to fill
     # the correct values in the final output. Every time we find a 1, we are going to stick
     # the vertical column of the interpolated data.
     local_target_hcoords_bitmask::T3
 
-    # Coefficients (WI1[, WI2]) used for the interpolation. Array of tuples.
-    interpolation_coeffs::T4
+    # Tuple of arrays of weights that performs horizontal interpolation (fixed the
+    # horizontal and target spaces). It contains 1 element for grids with one horizontal
+    # dimension, and 2 elements for grids with two horizontal dimensions.
+    local_horiz_interpolation_weights::T4
 
     # Local indices the element of each local_target_hcoords in the given topology
-    local_indices::T5
-
-    # Space over which the remapper is defined
-    space::T6
+    local_horiz_indices::T5
+
+    # Given the target_zcoords, vert_reference_coordinates contains the reference coordinate
+    # in the element. For center spaces, the reference coordinate is shifted to be in (0, 1)
+    # when the point is in the lower half of the cell, and in (-1, 0) when it is in the
+    # upper half. This shift is needed to directly use the reference coordinate in linear
+    # vertical interpolation. Array of tuples or Nothing.
+    vert_interpolation_weights::VERT_W
+
+    # Given the target_zcoords, vert_bounding_indices contain the vertical indices of the
+    # neighboring elements that are required for vertical interpolation.
+    # Array of tuples or Nothing.
+    vert_bounding_indices::VERT_IND
+
+    # Scratch space where we save the process-local interpolated values. We keep overwriting this
+    # to avoid extra allocations.
+    _local_interpolated_values::T8
+
+    # Scratch space where we save the process-local field value. We keep overwriting this to
+    # avoid extra allocations. Ideally, we wouldn't need this and we would use views for
+    # everything.
+    _field_values::T9
+
+    # Storage area where the interpolated values are saved. This is meaningful only for the
+    # root process and gets filled by a interpolate call.
+    _interpolated_values::T10
```

Comment:
This is some form of array, correct? If so, what's the dims of this array? (let's add the dims to the docs)

Reply:
Added.